### ISSUE TITLE
SetValueTo & ToObject<T> difference?

### ISSUE THREAD

**Issue by Ketan-Desai** [cemdervis/SharpConfig#54]
// Sample ini file

[Sample]
Name = Ketan

```
    class Sample
    {
        public string Name;
        public string Address;
    }
```

```
            var sample1 = new Sample();

            var cfg = Configuration.LoadFromFile(@"sample.ini");

            cfg["Sample"].SetValuesTo(sample1);

            Sample sample2 = cfg["Sample"].ToObject<Sample>();
```

This both yeilds the same result, then what is the difference between these functions?

Thanks

**Comment by Ketan-Desai** [comment]
and when v3.2.0 will be released?

**Comment by Ketan-Desai** [comment]
// Sample ini file

[Header]
Columns = { 1, 2, 3 }

```
var instance = headerSection.ToObject<Item>();

    public class Item
    {
        public string[] Columns;
    }
```

ToObject() is throwing exception...

Unhandled Exception: System.InvalidOperationException: To obtain an array value, use GetValueArray() instead of GetValue().
   at SharpConfig.Setting.GetValue(Type type)
   at SharpConfig.Section.SetValuesTo(Object obj)
   at SharpConfig.Section.ToObject[T]()

**Comment by cemdervis** [comment]
The difference is that SetValuesTo() sets values of a section **to an existing object**, while ToObject<T>() creates **a new object from** the section values.
3.2.0 will be released in 1-2 weeks.
I'll look into that error, thanks for reporting.

**Comment by cemdervis** [comment]
Fixed with 3.2.0.
